crypto/evp/evp_fetch.c

```diff
@@ -107,35 +117,14 @@ static int put_method_in_store(OPENSSL_CTX *libctx, void *store,
     return 0;
 }
 
-static void *construct_method(const char *algorithm_name,
-                              const OSSL_DISPATCH *fns, OSSL_PROVIDER *prov,
-                              void *data)
+static void *construct_method(const char *name, const OSSL_DISPATCH *fns,
+                              OSSL_PROVIDER *prov, void *data)
 {
     struct method_data_st *methdata = data;
-    void *method = NULL;
-    int nid = OBJ_sn2nid(algorithm_name);
-
-    if (nid == NID_undef) {
-        /* Create a new NID for that name on the fly */
-        ASN1_OBJECT tmpobj;
-
-        /* This is the same as OBJ_create() but without requiring a OID */
-        tmpobj.nid = OBJ_new_nid(1);
-        tmpobj.sn = tmpobj.ln = methdata->name;
-        tmpobj.flags = ASN1_OBJECT_FLAG_DYNAMIC;
-        tmpobj.length = 0;
-        tmpobj.data = NULL;
+    /* TODO(3.0) get rid of the need for legacy NIDs */
+    int legacy_nid = OBJ_sn2nid(name);
```

Comment:
Why is this needed? OBJ_sn2nid uses ASN1_OBJECT and I thought removing that dependency that was part of the point of this PR?

Reply:
The legacy NID needs to be preserved because of things like this:

https://github.com/openssl/openssl/blob/9acbe07d2300d34a7ea846d9756f33b4595e32fb/crypto/evp/evp_lib.c#L25-L37

Reply:
Of course, I answered a bit too quickly, and only your first question.  I'll take a closer look at what I just quoted.

Reply:
The above code only needs to work for implicit lookup ciphers so shouldn't impact the evp_fetch code.